server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -330,9 +330,7 @@ public DesiredBalance compute(
         iterations.inc(i);
 
         final var assignments = new HashMap<ShardId, ShardAssignment>();
```

Comment:
Maybe it's also worth consider to use `Maps.newHashMapWithExpectedSize` here

Reply:
++ right here it makes sense, thanks!